pkg/controllers/dynakube/phase_test.go

```diff
@@ -0,0 +1,167 @@
+package dynakube
+
+import (
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/scheme/fake"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/status"
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestActiveGatePhaseChanges(t *testing.T) {
+	dynakube := &dynatracev1beta1.DynaKube{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testName,
+			Namespace: testNamespace,
+		},
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			ActiveGate: dynatracev1beta1.ActiveGateSpec{Capabilities: []dynatracev1beta1.CapabilityDisplayName{dynatracev1beta1.KubeMonCapability.DisplayName}},
+		},
+	}
+	t.Run(" no activegate statefulsets in cluster -> deploying", func(t *testing.T) {
+		fakeClient := fake.NewClient()
+		controller := &Controller{
+			client:    fakeClient,
+			apiReader: fakeClient,
+		}
+		phase := controller.determineDynaKubePhase(dynakube)
+		assert.Equal(t, status.Deploying, phase)
+	})
+	t.Run("Error accessing k8s api", func(t *testing.T) {
```

Comment:
please use one style, each test has a different syle, Capitalise and lowercase mix.